ext/sockets/sockets.c

```diff
@@ -2315,7 +2310,7 @@ PHP_FUNCTION(socket_export_stream)
 	 * We don't want streams to actually *do* anything though, so don't give it
 	 * anything apart from the protocol */
 	if (protocol != NULL) {
-		stream = php_stream_xport_create(protocol, protocollen, 0, 0, NULL, NULL, NULL, NULL, NULL);
+		stream = php_stream_xport_create(protocol, strlen(protocol), 0, 0, NULL, NULL, NULL, NULL, NULL);
```

Comment:
IMHO keeping the variable makes sense, as the compiler can elide the call to ``strlen()`` when the value is constant.

Reply:
This function is inherently "heavy", and it's not a high-frequency operation, the calculation of the text length of the protocol isn't a key factor affecting performance. Therefore, I have no preference for the way of calculation. In the new commit, I've added a const modifier to the protocol and used `sizeof(protocol - 1)` to enhance readability and 100% ensure that it would be comfortable in terms of performance.